src/main/scala/com/typesafe/webdriver/Session.scala

```diff
@@ -24,13 +24,18 @@ class Session(wd: WebDriverCommands, sessionConnectTimeout: FiniteDuration)
   startWith(Uninitialized, None)
 
   when(Uninitialized) {
-    case Event(Connect, None) =>
-      wd.createSession().onComplete {
-        case Success(sessionId) => self ! SessionCreated(sessionId)
+    case Event(Connect(desiredCapabilities, requiredCapabilities), None) =>
+      val origSender = sender()
+      wd.createSession(desiredCapabilities,requiredCapabilities).onComplete {
+        case Success((sessionId, Right(capabilities))) => self.tell(SessionCreated(sessionId, capabilities), origSender)
+        case Success((sessionId, Left(error))) =>
+          log.error("Stopping due to not being able to establish a session with requested capabilities - web driver error - {}.",error)
+          origSender.tell(SessionAborted(sessionId,error), origSender)
+          stop()
         case Failure(_: ConnectionAttemptFailedException) =>
           log.debug("Initial connection attempt failed - retrying shortly.")
           context.system.scheduler.scheduleOnce(500.milliseconds) {
-            self ! Connect
+            self.tell(Connect(desiredCapabilities, requiredCapabilities), origSender)
```

Comment:
So we can pass in the original sender as the sender for the message. 
This lets us send a SessionAborted message back to the original sender when the required capabilities cannot be met even when there are multiple attempts at establishing the session.